spec/patterns/core_spec.rb

```diff
@@ -108,10 +108,99 @@
 
   context "when using comma separators and other regexp" do
 
+    let(:pattern) { '((a=(?<a>%{UNIXPATH})?|b=(?<b>%{UNIXPATH})?)(,\s)?)+' }
+
+    let(:grok) do
+      grok = LogStash::Filters::Grok.new("match" => ["message", pattern])
+      grok.register
+      grok
+    end
+
     let(:value) { 'a=/some/path, b=/some/other/path' }
 
+    it "should match both paths" do # but does not
+      event = build_event(value)
+      grok.filter(event)
+      expect( event.to_hash['a'] ).to eql '/some/path,'
+      expect( event.to_hash['b'] ).to be nil
+    end
+
+  end
+
+  context "dotted path" do
+
+    it "should match path containing ." do
+      expect(grok_match(pattern, '/some/./path/')).to pass
+      expect(grok_match(pattern, '/some/../path')).to pass
+      expect(grok_match(pattern, '/../.')).to pass
+      expect(grok_match(pattern, '/.')).to pass
+      expect(grok_match(pattern, '/..')).to pass
+      expect(grok_match(pattern, '/...')).to pass
+
+      expect(grok_match(pattern, 'a/./b/c')).to pass
+      expect(grok_match(pattern, ',/.')).to pass
+      expect(grok_match(pattern, '+/.../')).to pass
+    end
+
+    it "should match path starting with ." do
+      expect(grok_match(pattern, '../0')).to pass
+      expect(grok_match(pattern, './~')).to pass
+      expect(grok_match(pattern, '.../-')).to pass
+      expect(grok_match(pattern, './')).to pass
+      expect(grok_match(pattern, './,')).to pass
+      expect(grok_match(pattern, '../')).to pass
+      expect(grok_match(pattern, '.a/')).to pass
+      expect(grok_match(pattern, '.~/')).to pass
+    end
```

Comment:
we should beware that these don't actually pass. the matcher will try to match `${UNIXPATH}` to any part of that string. For example in `./~`, the test passes since the match is `/~`.
When tested in the context of a full string, we can see this failing:

```
❯ bin/logstash -e 'filter { grok { match => { "message" => [ "a %{UNIXPATH:path} b" ] }}}'               
a ./~ b
{
          "host" => "joaos-mbp.lan",
          "type" => "stdin",
      "@version" => "1",
    "@timestamp" => 2020-11-30T09:57:26.213Z,
          "tags" => [
        [0] "_grokparsefailure"
    ],
       "message" => "a ./~ b"
}
```

Reply:
thanks, updated specs to do exact matches and also spec-ed out these with some partial matching ...